core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/filter/FilterPredicate.java

```diff
@@ -89,24 +98,66 @@ private boolean evaluateFilterTimestamp(Tuple inputTuple) {
     }
 
 
+
+
     private static <T extends Comparable<T>> boolean evaluateFilter(T value, T compareToValue, ComparisonType comparisonType) {
-        if (value == null) {
-            return compareToValue == null;
-        }
-        int compareResult = value.compareTo(compareToValue);
+        int compareResult;
         switch (comparisonType) {
+            case Xnull:
+                 if(value==null){
```

Comment:
we have 7 identical code for null check, can't we refactor them to just do one null check?